python/psautohint/otfFont.py

```diff
@@ -440,17 +437,70 @@ def makeRelativeCTArgs(argList, curX, curY):
     return argList, newCurX, newCurY
 
 
-def convertBezToT2(bezString):
+def build_hint_order(hints):
+    # MM hints have duplicate hints. We want to return a list of indices into
+    # the original unsorted and unfiltered list. The list should be sorted, and
+    # should filter out duplicates
+
+    numHints = len(hints)
+    index_list = list(range(numHints))
+    hint_list = list(zip(hints, index_list))
+    hint_list.sort()
+    new_hints = [hint_list[i] for i in range(1, numHints)
+                 if hint_list[i][0] != hint_list[i-1][0]]
+    new_hints = [hint_list[0]] + new_hints
+    hints, hint_order = list(zip(*new_hints))
```

Comment:
I believe the comment means: "calling hint_order.index(index_from_bez) will yield the hint index after sorting".
I first interpreted it backwards: "hint_order[index_from_bez] will yield the hint index after sorting".
Seems misleading; better clarify?

Reply:
Will do this, and your other suggestion. FYI, I am working on a new commit to remove hint pairs that are incorrect in the MM hinted region fonts.